src/main/java/org/broadinstitute/dsm/model/filter/FilterFactory.java

```diff
@@ -0,0 +1,76 @@
+package org.broadinstitute.dsm.model.filter;
+
+import java.util.Collections;
+import java.util.Objects;
+
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.dsm.model.filter.participant.EmptyFilterParticipantList;
+import org.broadinstitute.dsm.model.filter.participant.ManualFilterParticipantList;
+import org.broadinstitute.dsm.model.filter.participant.QuickFilterParticipantList;
+import org.broadinstitute.dsm.model.filter.participant.SavedFilterParticipantList;
+import org.broadinstitute.dsm.model.filter.tissue.ManualFilterTissueList;
+import org.broadinstitute.dsm.model.filter.tissue.SavedFiltersTissueList;
+import org.broadinstitute.dsm.model.filter.tissue.SearchBarFilterTissueList;
+import org.broadinstitute.dsm.statics.DBConstants;
+import org.broadinstitute.dsm.statics.RequestParameter;
+import org.broadinstitute.dsm.statics.RoutePath;
+import spark.QueryParamsMap;
+import spark.Request;
+
+public class FilterFactory {
+
+
+    public static Filterable of(Request request) {
+        QueryParamsMap queryParams = Objects.requireNonNull(request).queryMap();
+        String parent = queryParams.get(DBConstants.FILTER_PARENT).value();
+        if (StringUtils.isBlank(parent)) throw new IllegalArgumentException("parent cannot be empty");
+        String[] pathSegments = request.uri().split("/");
+        String lastSegment = pathSegments.length > 0 ? pathSegments[pathSegments.length-1] : "";
+        String jsonBody = request.body();
+        Filterable filterable = queryParamsMap -> Collections.emptyList();
+        switch (lastSegment) {
+            case RoutePath.APPLY_FILTER:
+                switch (parent) {
+                    case BaseFilter
+                            .PARENT_PARTICIPANT_LIST:
+                        if (StringUtils.isNotBlank(queryParams.get(RequestParameter.FILTER_NAME).value())) {
+                            filterable = new QuickFilterParticipantList();
+                        } else if (StringUtils.isNotBlank(queryParams.get(RequestParameter.FILTERS).value())){
+                            filterable = new SavedFilterParticipantList();
+                        } else {
+                            filterable = new EmptyFilterParticipantList();
+                        }
+                        break;
+                    case BaseFilter.TISSUE_LIST_PARENT:
+                        String filterQuery = queryParams.get("filterQuery").value();
+                        if (StringUtils.isNotBlank(filterQuery)) {
+                            filterable = new SearchBarFilterTissueList();
```

Comment:
Where do we filter the participant list search bar ?

Reply:
Good question, I meant this when I mentioned poor design of filtering overall, frontend side is such organized that search bar filtering for tissue list parent comes this way, and search bar filtering for participant list parent comes totally different way.